### DISCUSSION TITLE
Connect sets null unique field when new row tries to use same value

### DISCUSSION THREAD

**Discussion by MilenaTar** [prisma/prisma#18962] [Q&A]
```
model User {
 id             String       @id @default(cuid())
lead         Lead?
}

model Lead {
  id                        String            @id @default(cuid())
  user                    User?           @relation(fields: [userId], references: [id], onDelete: SetNull)
  userId                 String?       @unique
}
```
There is lead with connected user, when I try to connect userId to new lead, the old one is set to null. Is there any way to change this and get an error?

**Comment by nurul3101** [comment]
Hello @MilenaTar 👋 

Can a single user have multiple leads? If yes, then you need to create a one-to-many relation instead of the current one-to-one.

**Reply by MilenaTar** [reply]
Hi @nurul3101, 
Thanks for reply. No user cant have multiple leads.

**Reply by nurul3101** [reply]
Can you share the query through which you are connecting records?
Can you try `onUpdate: Restrict` [referential action](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions#restrict) and check?

**Reply by nurul3101** [reply]
@MilenaTar If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by MilenaTar** [reply]
> @MilenaTar If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved.
> 
> Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

I'm sorry for late response. `onUpdate: Restrict` doesn't help me, I've changed db structure

**Comment by MilenaTar** [comment]
There is still problem with prisma.update() case when  connect sets null unique field when new row tries to use same value

**Reply by nurul3101** [reply]
Can you share the query that you are invoking?

**Reply by MilenaTar** [reply]
await this.prisma.lead.update({
      data: {
        user: {
          connect: {
            id: userId
          }
        }
      },
      where: {
        id: lead.id
      }
    });

where userId is already connected to another lead.
